src/frontends/tensorflow/src/helper_ops/block_lstm.hpp

```diff
@@ -0,0 +1,143 @@
+        // 3) h_prev A Tensor. Must have the same type as x. Initial output of cell (to be used for peephole).
+        // 4) w	A Tensor. Must have the same type as x. The weight matrix.
+        // 5) wci A Tensor. Must have the same type as x. The weight matrix for input gate peephole connection.
+        // 6) wcf A Tensor. Must have the same type as x. The weight matrix for forget gate peephole connection.
+        // 7) wco A Tensor. Must have the same type as x. The weight matrix for output gate peephole connection.
+        // 8) b	A Tensor. Must have the same type as x. The bias vector.
+        //
+        // Attributes:
+        // forget_bias	An optional float. Defaults to 1. The forget gate bias.
+        // cell_clip	An optional float. Defaults to 3. Value to clip the 'cs' value to.
+        // use_peephole	An optional bool. Defaults to False. Whether to use peephole weights.
+        //
+        // Outputs:
+        // 0) i A Tensor. Has the same type as x.
+        // 1) cs A Tensor. Has the same type as x.
+        // 2) f A Tensor. Has the same type as x.
+        // 3) o A Tensor. Has the same type as x
+        // 4) ci A Tensor. Has the same type as x.
+        // 5) co A Tensor. Has the same type as x.
+        // 6) h A Tensor. Has the same type as x.
+
+        // extract time_len, batch_size
+        auto time_len = ov::Dimension::dynamic();
+        auto batch_size = ov::Dimension::dynamic();
+        auto x_shape = get_input_partial_shape(1);
+        auto x_type = get_input_element_type(1);
+        auto x_rank = x_shape.rank();
+        if (x_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                x_rank.get_length() == 3,
+                "Internal error in OpenVINO TensorFlow Frontend: input data for BlockLSTM must be of rank equal to 3.");
+            time_len = x_shape[0].is_static() ? x_shape[0].get_length() : time_len;
+            batch_size = x_shape[1].is_static() ? x_shape[1].get_length() : batch_size;
+        }
+
+        // extract hidden_size
+        auto w_shape = get_input_partial_shape(4);
+        auto w_rank = w_shape.rank();
+        auto b_shape = get_input_partial_shape(8);
+        auto b_rank = b_shape.rank();
+        if (w_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                w_rank.get_length() == 2,
+                "Internal error in OpenVINO TensorFlow Frontend: weights for BlockLSTM must be of rank equal to 2.");
+            m_hidden_size = w_shape[1].is_static() ? w_shape[1].get_length() / 4 : ov::Dimension::dynamic();
+        }
+        if (b_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                b_rank.get_length() == 1,
+                "Internal error in OpenVINO TensorFlow Frontend: weights for BlockLSTM must be of rank equal to 2.");
+            m_hidden_size = b_shape[0].is_static() ? b_shape[0].get_length() / 4 : ov::Dimension::dynamic();
```

Comment:
there is a change to re-write m_hidden_size if it was already set to static
maybe we can add additional check here
if (b_rank.is_static() && m_hidden_size.is_dynamic()) {

Reply:
In the current version, I check the rank length of `bias` this way.
In your case, we miss this check if the rank is static and the dimension is dynamic.

Reply:
I just highlighted a mistake, please fix it
maybe this way or any other
if (b_rank.is_static()) {
 check(...);
  if (m_hidden_size.is_dynamic()) {
   m_hidden_size = ...;

Reply:
Fixed.